src/script.js

```diff
@@ -0,0 +1,96 @@
+// Define the task array with sample tasks
+const tasks = [
+  {
+    description: 'Task 1',
+    completed: false,
+    index: 0,
+  },
+  {
+    description: 'Task 2',
+    completed: true,
+    index: 1,
+  },
+  {
+    description: 'Task 3',
+    completed: false,
+    index: 2,
+  },
+];
+
+// Define the function to add new task
+const addTask = (description) => {
+  const task = {
+    description,
+    completed: false,
+    index: tasks.length,
+  };
+  tasks.push(task);
+  return task;
+};
+
+// Define the function to toggle task completion status
+const toggleTaskCompletion = (id) => {
+  const task = tasks[id];
+  task.completed = !task.completed;
+  return task;
+};
+
+// Define the function to remove a task
+const removeTask = (id) => {
+  tasks.splice(id, 1);
+};
+
+// Define the function to populate the task list
+const populateTasks = () => {
+  const parentElement = document.getElementById('taskList');
+  parentElement.innerHTML = '';
+  tasks.forEach((task) => {
+    const { description, completed, index: id } = task;
+    const innerHTML = `
+      <p>${description}</p>
+      <p>${completed ? 'completed' : 'not completed'}</p>
+      <button class="removeButton">Remove</button>
+      <input type="checkbox" ${completed ? 'checked' : ''}>
+    `;
+    const newTodoListElement = document.createElement('li');
+    Object.assign(newTodoListElement, { innerHTML, id });
+    parentElement.appendChild(newTodoListElement);
+  });
+};
+
+// Add event listener for form submission
+const form = document.getElementById('addTaskForm');
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  const input = document.getElementById('newTaskInput');
+  const description = input.value.trim();
+  if (description) {
+    addTask(description);
+    input.value = '';
+    populateTasks();
+  }
+});
```

Comment:
- [ ] At the moment, the add task functionality is not working which makes it almost impossible to test the other functionalities. Please kindly help fix this so it can be properly tested.